src/main/java/com/baro/template/application/TemplateService.java

```diff
@@ -0,0 +1,29 @@
+package com.baro.template.application;
+
+import com.baro.template.application.dto.FindTemplateQuery;
+import com.baro.template.application.dto.FindTemplateResult;
+import com.baro.template.domain.TemplateRepository;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@RequiredArgsConstructor
+@Transactional
+@Service
+public class TemplateService {
+
+    private final TemplateRepository repository;
+
+    public List<FindTemplateResult> findTemplates(FindTemplateQuery query) {
```

Comment:
🌘
@Transactional(readOnly = true) 옵션을 지정해도 좋아보여요!
```suggestion
    @Transactional(readOnly = true)
    public List<FindTemplateResult> findTemplates(FindTemplateQuery query) {
```